prow/bump.sh

```diff
@@ -49,17 +53,77 @@ if ! ($SED --version 2>&1 | grep -q GNU); then
   exit 1
 fi
 
-new_version="v$(date -u '+%Y%m%d')-$(git describe --tags --always --dirty)"
-echo -e "version: $(color-version ${new_version})" >&2
-if [[ "${new_version}" == *-dirty ]]; then
-  echo -e "$(color-error ERROR): uncommitted changes to repo" >&2
-  echo "  Fix with git commit" >&2
+cd "$(dirname "${BASH_SOURCE}")"
+
+usage() {
+  echo "Usage: "$(basename "$0")" [--push || --list || vYYYYMMDD-deadbeef] [image subset...]" >&2
   exit 1
+}
+
+if [[ -n "${GOOGLE_APPLICATION_CREDENTIALS:-}" ]]; then
+  echo "Detected GOOGLE_APPLICATION_CREDENTIALS, activating..." >&2
+  gcloud auth activate-service-account --key-file="$GOOGLE_APPLICATION_CREDENTIALS"
+  gcloud auth configure-docker
+  cmd="--push"
+elif [[ $# == 0 ]]; then
+  usage
+else
+  cmd="$1"
+  shift
 fi
 
-cd "$(dirname "${BASH_SOURCE}")"
+if [[ "$cmd" == "--push" ]]; then
+  new_version="v$(date -u '+%Y%m%d')-$(git describe --tags --always --dirty)"
+  echo -e "version: $(color-version ${new_version})" >&2
+  if [[ "${new_version}" == *-dirty ]]; then
+    echo -e "$(color-error ERROR): uncommitted changes to repo" >&2
+    echo "  Fix with git commit" >&2
+    exit 1
+  fi
+  echo -e "Pushing $(color-version ${new_version}) via $(color-target //prow:release-push --platforms=@io_bazel_rules_go//go/toolchain:linux_amd64) ..." >&2
+  bazel run //prow:release-push --platforms=@io_bazel_rules_go//go/toolchain:linux_amd64
+elif [[ "$cmd" == "--list" ]]; then
+  # TODO(fejta): figure out why the timestamp on all these image is 1969...
+  # Then we'll be able to just sort them.
+  options=(
+    $(gcloud container images list-tags gcr.io/k8s-prow/plank --filter="tags ~ ^v|,v" --format='value(tags)' \
```

Comment:
maybe a concern: what if someone only bumped plank...

Reply:
What do you mean? We always build and push all images. 

It is fine to only update the deployment for plank (`bump.sh --list plank`)

Reply:
ah ok now we got rid of the Makefile, so :+1: